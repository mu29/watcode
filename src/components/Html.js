/* eslint-disable react/no-danger */
import React from 'react'
import { PropTypes } from 'prop-types'
import { Helmet } from 'react-helmet'

const Html = ({ styles, state, children }) => {
  const helmet = Helmet.renderStatic()
  const htmlAttrs = helmet.htmlAttributes.toComponent()
  const bodyAttrs = helmet.bodyAttributes.toComponent()

  return (
    <html lang="ko" { ...htmlAttrs }>
      <head>
        { helmet.title.toComponent() }
        { helmet.meta.toComponent() }
        { helmet.link.toComponent() }
        <meta charSet="utf8" />
        <link href="/style.css" rel="stylesheet" type="text/css" />
        <link href="//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSans-kr.css" rel="stylesheet" type="text/css" />
        <script defer src="https://use.fontawesome.com/releases/v5.0.13/js/all.js" integrity="sha384-xymdQtn1n3lH2wcu0qhcdaOpQwyoarkgLVxC/wZ5q7h9gHtxICrpcaSUfygqZGOe" crossOrigin="anonymous" />
        { styles }
      </head>
      <body { ...bodyAttrs }>
        <main id="app">{ children }</main>
        <script dangerouslySetInnerHTML={{ __html: state }} />
        <script src="/client.js" />
      </body>
    </html>
  )
}

Html.propTypes = {
  styles: PropTypes.node.isRequired,
  state: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default Html
