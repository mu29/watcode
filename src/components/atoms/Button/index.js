import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { Link } from 'react-router-dom'
import {
  width,
  height,
  padding,
  fontFamily,
  fontSize,
  color,
  primaryColor,
  backgroundColor,
  border,
  borderRadius,
} from 'services/style'

const styles = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: ${width};
  height: ${height};
  padding: ${padding};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  color: ${color};
  background-color: ${backgroundColor};
  border: ${border};
  border-radius: ${borderRadius};
  cursor: pointer;
  transition: color 0.3s;

  &:hover {
    color: ${primaryColor};
  }

  &:active, &:focus, &:hover {
    outline: none;
  }
`

const StyledLink = styled(({
  to, ...props
}) => <Link to={ to } { ...props } />)`${styles}`

const Anchor = styled.a`${styles}`

const StyledButton = styled.button`${styles}`

const Button = ({
  to, href, ...props
}) => {
  if (to) {
    return <StyledLink to={ to } { ...props } />
  } else if (href) {
    return <Anchor href={ href } { ...props } />
  }
  return <StyledButton { ...props } />
}

Button.propTypes = {
  to: PropTypes.string,
  href: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
  padding: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  block: PropTypes.bool,
  font: PropTypes.string,
  fontSize: PropTypes.number,
  color: PropTypes.string,
  primaryColor: PropTypes.string,
  backgroundColor: PropTypes.string,
  border: PropTypes.string,
  borderRadius: PropTypes.number,
}

Button.defaultProps = {
  to: undefined,
  href: undefined,
  width: undefined,
  height: 48,
  padding: '0 24',
  block: false,
  font: 'default',
  fontSize: 16,
  color: 'white.default',
  primaryColor: 'yellow.default',
  backgroundColor: 'gray.90',
  border: undefined,
  borderRadius: 0,
}

export default Button
