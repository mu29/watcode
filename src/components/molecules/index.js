export { default as Field } from './Field'
export { default as IconButton } from './IconButton'
export { default as Tab } from './Tab'
