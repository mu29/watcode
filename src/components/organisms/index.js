export { default as ArtworkList } from './ArtworkList'
export { default as Header } from './Header'
export { default as LoginForm } from './LoginForm'
export { default as ReduxField } from './ReduxField'
export { default as RegisterForm } from './RegisterForm'
export { default as SearchBar } from './SearchBar'
