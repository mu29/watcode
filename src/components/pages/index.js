export { default as HomePage } from './HomePage'
export { default as RankingPage } from './RankingPage'
export { default as BookmarkPage } from './BookmarkPage'
