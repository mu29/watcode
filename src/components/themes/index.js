export { default as basicTheme } from './basic'
