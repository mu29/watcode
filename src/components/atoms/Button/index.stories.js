import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import Button from '.'

storiesOf('Atoms/Button', module)
  .add('기본', () => (
    <Button>확인</Button>
  ))
  .add('배경 색', () => (
    <Button backgroundColor="red.default">확인</Button>
  ))
  .add('테두리 색', () => (
    <Button color="gray.100" backgroundColor="white.default" border="red.default">확인</Button>
  ))
  .add('너비', () => (
    <Button width={ 144 }>확인</Button>
  ))
  .add('높이', () => (
    <Button height={ 72 }>확인</Button>
  ))
  .add('너비 + 높이', () => (
    <Button width={ 144 } height={ 72 }>확인</Button>
  ))
  .add('블록', () => (
    <Button block>확인</Button>
  ))
  .add('onClick', () => (
    <Button onClick={ action('button-click') }>확인</Button>
  ))
