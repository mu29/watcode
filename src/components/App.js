import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { injectGlobal, ThemeProvider } from 'styled-components'
import { authorizeActions, readBookmarksActions } from 'store/actions'
import { auth } from 'services/firebase'
import { basicTheme } from './themes'
import Router from './Router'

/* eslint-disable no-unused-expressions */
injectGlobal`
  * {
    box-sizing: border-box;
  }
  body {
    margin: 0;
  }
`

export default class App extends React.Component {
  static propTypes = {
    store: PropTypes.shape({
      dispatch: PropTypes.func,
      getState: PropTypes.func,
    }).isRequired,
  }

  componentWillMount() {
    const { store } = this.props
    auth.onAuthStateChanged((user) => {
      store.dispatch(authorizeActions.request({ user }))
      store.dispatch(readBookmarksActions.request())
    })
  }

  render() {
    const state = this.props.store.getState()
    const isLoggedIn = state.auth.user.email !== undefined
    return (
      <React.Fragment>
        <Helmet titleTemplate="watcode">
          <title>왓코드 - 오늘 볼 만화 추천해줘!</title>
          <meta charSet="UTF-8" />
          <meta name="description" content="" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <meta property="og:site_name" content="watcode" />
        </Helmet>
        <ThemeProvider theme={ basicTheme }>
          <Router isLoggedIn={ isLoggedIn } />
        </ThemeProvider>
      </React.Fragment>
    )
  }
}
