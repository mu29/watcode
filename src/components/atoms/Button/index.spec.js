import React from 'react'
import { shallow } from 'enzyme'
import { basicTheme } from 'components/themes'
import 'jest-styled-components'
import Button from '.'

const onClick = jest.fn()

const wrap = (props = {}) =>
  shallow(<Button theme={ basicTheme } onClick={ onClick } { ...props } />)

describe('<Button />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })

  it('자식 컴포넌트가 들어온 경우 그려준다', () => {
    const wrapper = wrap({ children: 'test' })
    expect(wrapper).toMatchSnapshot()
  })

  it('임의의 속성이 들어온 경우 그려준다', () => {
    const wrapper = wrap({ name: 'foo' })
    expect(wrapper.props().name).toEqual('foo')
  })

  it('기본으로 button 태그를 그려준다', () => {
    const wrapper = wrap().dive()
    expect(wrapper.find('button')).toHaveLength(1)
  })

  it('href 속성이 들어온 경우 anchor 태그를 그려준다', () => {
    const wrapper = wrap({ href: 'test' }).dive()
    expect(wrapper.find('a')).toHaveLength(1)
  })

  it('to 속성이 들어온 경우 Link 컴포넌트를 그려준다', () => {
    const wrapper = wrap({ to: 'test' }).dive()
    expect(wrapper.find('Link')).toHaveLength(1)
  })

  it('누르면 onClick 함수를 호출한다', () => {
    onClick.mockClear()
    const wrapper = wrap()
    expect(onClick).not.toBeCalled()
    wrapper.simulate('click')
    expect(onClick).toBeCalled()
  })
})
