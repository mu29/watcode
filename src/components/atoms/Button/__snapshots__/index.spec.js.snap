// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> 자식 컴포넌트가 들어온 경우 그려준다 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3rem;
  padding: 0rem 1.5rem;
  font-family: Spoqa Han Sans,Helvetica Neue,Nanum Gothic;
  font-size: 1rem;
  color: #FFF;
  background-color: #424242;
  border: none;
  border-radius: 0.0625rem;
  cursor: pointer;
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
}

.c0:hover {
  color: #F9D14C;
}

.c0:active,
.c0:focus,
.c0:hover {
  outline: none;
}

<button
  className="c0"
  color="white.default"
  fontSize={16}
  height={48}
  onClick={[MockFunction]}
>
  test
</button>
`;

exports[`<Button /> 정상적으로 렌더링된다 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3rem;
  padding: 0rem 1.5rem;
  font-family: Spoqa Han Sans,Helvetica Neue,Nanum Gothic;
  font-size: 1rem;
  color: #FFF;
  background-color: #424242;
  border: none;
  border-radius: 0.0625rem;
  cursor: pointer;
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
}

.c0:hover {
  color: #F9D14C;
}

.c0:active,
.c0:focus,
.c0:hover {
  outline: none;
}

<button
  className="c0"
  color="white.default"
  fontSize={16}
  height={48}
  onClick={[MockFunction]}
/>
`;
